File: core/src/main/java/io/github/bindglam/neko/utils/Plugins.java
package io.github.bindglam.neko.utils;

import io.github.bindglam.neko.Neko;
import io.github.bindglam.neko.NekoPlugin;
import org.bukkit.plugin.java.JavaPlugin;
import org.bukkit.plugin.Plugin;
import org.jetbrains.annotations.NotNull;

import java.util.logging.Logger;

public final class Plugins {
    public static JavaPlugin plugin(NekoPlugin nekoPlugin) {
        return (JavaPlugin) nekoPlugin;
    }

    public static Logger logger() {
        return plugin(Neko.plugin()).getLogger();
    }

    private Plugins() {
    }
}

File: core/src/main/kotlin/io/github/bindglam/neko/utils/Plugins.kt
package io.github.bindglam.neko.utils

import io.github.bindglam.neko.Neko
import io.github.bindglam.neko.NekoPlugin
import org.bukkit.plugin.java.JavaPlugin

fun NekoPlugin.plugin() = this as JavaPlugin

fun logger() = Neko.plugin().plugin().logger